tests/image/image_common.h

```diff
@@ -421,6 +420,383 @@ cl::sycl::vector_class<cl::sycl::byte> get_image_host(
   return imageHost;
 }
 
+template <int dims>
+struct image_kernel_read;
+template <int dims>
+struct image_kernel_write;
+class empty_kernel {
+ public:
+  void operator()() const {}
+};
+
+template <int dims>
+class image_api_check {
+  int calc_numElems(util::logger &log, cl::sycl::range<dims> &r,
+                    cl::sycl::range<dims - 1> *p) const {
+    auto numElems = 1;
+    switch (dims) {
+      case 1:
+        log.note("Testing image combination: dims[%d], range[%d]", dims, r[0]);
+        numElems = r[0];
+        break;
+      case 2:
+        log.note(
+            "Testing image combination: dims[%d], range[%d, %d], pitch[%d]",
+            dims, r[0], r[1], (p != nullptr));
+        numElems = r[0] * r[1];
+        break;
+      case 3:
+        log.note(
+            "Testing image combination: dims[%d], range[%d, %d, %d], pitch[%d]",
+            dims, r[0], r[1], r[2], (p != nullptr));
+        numElems = r[0] * r[1] * r[2];
+        break;
+      default:
+        break;
+    }
+    return numElems;
+  }
+
+  template <typename AllocatorT = cl::sycl::image_allocator>
+  void check_api(util::logger &log, cl::sycl::image<dims, AllocatorT> &img,
+                  cl::sycl::range<dims> &r, cl::sycl::range<dims - 1> *p,
+                  int numElems, int elementSize,
+                  cl::sycl::vector_class<cl::sycl::byte> &finalData) const {
+
+    // Check get_range()
+    if (dims == 3) {
+      if ((!CHECK_VALUE_SCALAR(log, img.get_range()[0], r.get(0))) ||
+          (!CHECK_VALUE_SCALAR(log, img.get_range()[1], r.get(1))) ||
+          (!CHECK_VALUE_SCALAR(log, img.get_range()[2], r.get(2)))) {
+        FAIL(log, "Ranges are not the same.");
+      }
+    }
+    if (dims == 2) {
+      if ((!CHECK_VALUE_SCALAR(log, img.get_range()[0], r.get(0))) ||
+          (!CHECK_VALUE_SCALAR(log, img.get_range()[1], r.get(1)))) {
+        FAIL(log, "Ranges are not the same.");
+      }
+    }
+    if (dims == 1) {
+      if ((!CHECK_VALUE_SCALAR(log, img.get_range()[0], r.get(0)))) {
+        FAIL(log, "Ranges are not the same.");
+      }
+    }
+
+    // Check get_pitch()
+    if (!image_generic<dims, AllocatorT>::compare_pitch(log, img, p)) {
+      FAIL(log, "Pitches are not the same.");
+    }
+
+    // Check get_size()
+    if (img.get_size() < (numElems * elementSize)) {
+      cl::sycl::string_class message =
```

Comment:
I think you can use `std::string` for non-API related stuff.

Reply:
Changed to std::string.